account_journal_general_sequence/models/account_move.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Moduon
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    _sql_constraints = [
+        (
+            "entry_number_unique",
+            "UNIQUE(entry_number, company_id)",
+            "Entry number must be unique per company.",
+        ),
+    ]
+
+    entry_number = fields.Char(
+        index=True,
+        readonly=True,
+        tracking=True,
+        store=True,
+        compute="_compute_entry_number",
+        help="Automatic numbering, based on journal configuration.",
+    )
+
+    @api.depends("state")
+    def _compute_entry_number(self):
+        """Assign an entry number when posting."""
+        canceled = self.filtered_domain(
+            [("state", "=", "cancel"), ("entry_number", "!=", False)]
+        )
+        canceled.entry_number = False
+        if canceled:
+            no_gap_seqs = canceled.mapped(
+                "journal_id.entry_number_sequence_id"
+            ).filtered_domain([("implementation", "=", "no_gap")])
+            if no_gap_seqs:
+                _logger.warning(
```

Comment:
Curiosity: If the sequence does not admit gaps, do you allow it to be False (leaving a gap in the sequence) and only outputting the corresponding warning in the log?

Reply:
Yes. This is because the user can do nothing about this problem when cancelling a move, so I don't see reasons to bother them with the alert. However, before filing taxes with the tax authorities, they can use the provided renumbering wizard. That wizard will remove all gaps.

Do you think the user should be warned in this moment? cc @rafaelbn

Reply:
I think is correct and all my test looks 👍 😄 Just I did a comment around the renumber internally and we will update with this tomorrow.

@DarioLodeiros the sequence doesn't admit gaps but with manual actions because notation errors (human) could happens so we should renumber.